astropy/coordinates/representation.py

```diff
@@ -1837,10 +1837,15 @@ def transform(self, matrix):
         NotImplementedError
 
         """
-        raise NotImplementedError(
-            "Radial representations cannot be transformed "
-            "by a Cartesian matrix."
-        )
+        not_diagonal = np.any(np.count_nonzero(x - np.diag(np.diagonal(x))))
+        if not_diagonal:
+            raise NotImplementedError(
+                "Radial representations cannot be transformed"
+                " by a non-diagonal Cartesian matrix.")
+
+        # take the matrix transpose so that a (M,N,N) becomes an (N,N,M) matrix
+        # matrix and the [0,0] element broadcasts along the M axis.
+        return self * matrix.T[0, 0]
```

Comment:
I'd suggest writing this as `self * matrix[..., 0, 0]` (and then the comment is not needed any more)

Reply:
💡